freqtrade/freqai/data_kitchen.py

```diff
@@ -206,11 +206,15 @@ def filter_features(
 
         drop_index = pd.isnull(filtered_df).any(axis=1)  # get the rows that have NaNs,
         drop_index = drop_index.replace(True, 1).replace(False, 0)  # pep8 requirement.
+        ft_params = self.freqai_config["feature_parameters"]
         if (training_filter):
-            const_cols = list((filtered_df.nunique() == 1).loc[lambda x: x].index)
-            if const_cols:
-                filtered_df = filtered_df.filter(filtered_df.columns.difference(const_cols))
-                logger.warning(f"Removed features {const_cols} with constant values.")
+            if not ft_params.get(
```

Comment:
I think we still want to check for constant columns when we filter for training data (even when `principal_component_analysis` is true). This occurs *before* PCA, and we want to feed the non constant cols into PCA. 

Correct me if I am missing something.